### ISSUE TITLE
Update numpydoc

### ISSUE THREAD

**Issue by kohr-h** [odlgroup/odl#1111]
Seems they removed their homebrew `linkcode.py`, and the current master seems to be less broken for us than what we use now. Last time I tested it, links to attributes worked again, but inherited attributes didn't.

I think I'll do the following:
- Make a rebased branch in our fork of `numpydoc`
- Make a PR here that updates the submodule to the branch and test it.
- Rebase the master branch our fork of `numpydoc`
- Update the PR
- Merge the PR

Hopefully that doesn't break stuff too much in the meanwhile :-).

**Comment by adler-j** [comment]
I agree that this should be done. Long term we need to do some significant improvements to the doc. For one, I think we should drop the "separate file for each method" and change that to some module level stuff.
